src/test/shell/bazel/bazel_coverage_cc_test_llvm.sh

```diff
@@ -218,5 +218,194 @@ end_of_record"
   assert_equals "$expected_result" "$(cat $(get_coverage_file_path_from_test_log))"
 }
 
+function setup_external_cc_target() {
+  cat > WORKSPACE <<'EOF'
+local_repository(
+    name = "other_repo",
+    path = "other_repo",
+)
+EOF
+
+  cat > BUILD <<'EOF'
+cc_library(
+    name = "b",
+    srcs = ["b.cc"],
+    hdrs = ["b.h"],
+    visibility = ["//visibility:public"],
+)
+EOF
+
+  cat > b.h <<'EOF'
+int b(bool what);
+EOF
+
+  cat > b.cc <<'EOF'
+int b(bool what) {
+  if (what) {
+    return 1;
+  } else {
+    return 2;
+  }
+}
+EOF
+
+  mkdir -p other_repo
+  touch other_repo/WORKSPACE
+
+  cat > other_repo/BUILD <<'EOF'
+cc_library(
+    name = "a",
+    srcs = ["a.cc"],
+    hdrs = ["a.h"],
+    deps = ["@//:b"],
+)
+
+cc_test(
+    name = "t",
+    srcs = ["t.cc"],
+    linkstatic = True,
+    deps = [":a"],
+)
+EOF
+
+  cat > other_repo/a.h <<'EOF'
+int a(bool what);
+EOF
+
+  cat > other_repo/a.cc <<'EOF'
+#include "a.h"
+#include "b.h"
+
+int a(bool what) {
+  if (what) {
+    return b(what);
+  } else {
+    return 1 + b(what);
+  }
+}
+EOF
+
+  cat > other_repo/t.cc <<'EOF'
+#include <stdio.h>
+#include "a.h"
+
+int main(void) {
+  a(true);
+}
+EOF
+}
+
+function test_external_cc_target_can_collect_coverage() {
+  local -r llvm_profdata="/usr/bin/llvm-profdata-9"
+  if [[ ! -x ${llvm_profdata} ]]; then
+    return
+  fi
+
+  local -r clang="/usr/bin/clang-9"
```

Comment:
Do we need to constrain to a single version? Will your tests not work for multiple versions?

Reply:
I chose the same constraint as in the other tests that assert something about lcov output. If I use no constraint, the tests run into the failures we saw previously. They also pass locally with clang 14 and gcc 11. Do you want me to use other versions?

Reply:
If we could test against both versions then we're less likely to end up in a situation where nothing is being run for our automated tests.

I'm not exactly satisfied, and maybe in the future I'll spend some time trying to improve it, but for now if it's possible to increase robustness in this manner, I'm inclined to take it.

Reply:
@c-mita Turns out that LLVM <9, 10, and 11 don't work, 9 and >11 do work. I added a version check and also replaced the hard-coded versions in other tests in the file with it.